src/usr/local/sbin/openvpn.attributes.sh

```diff
@@ -18,16 +18,41 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
+lockfile="/tmp/ovpn_${dev}_${username}_${trusted_port}.lock"
+rulesfile="/tmp/ovpn_${dev}_${username}_${trusted_port}.rules"
+anchorname="openvpn/${dev}_${username}_${trusted_port}"
+
 if [ "$script_type" = "client-connect" ]; then
-	if [ -f /tmp/$common_name ]; then
-		/bin/cat /tmp/$common_name > $1
-		/bin/rm /tmp/$common_name
-	fi
+    while [ -f "${lockfile}" ];
+    do
+        sleep 1
```

Comment:
There should be some kind of alternate escape plan for this loop in case another process gets stuck.